bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -384,50 +385,69 @@ private static boolean isConvertibleToRecordType(Object sourceValue, BRecordType
         MapValueImpl sourceMapValueImpl = (MapValueImpl) sourceValue;
         for (Map.Entry targetTypeEntry : targetFieldTypes.entrySet()) {
             String fieldName = targetTypeEntry.getKey().toString();
+            String fieldNameLong;
+            if (varName == null) {
```

Comment:
Shall we move this logic to a method and reuse? It is duplicated in line 409 as well.